metagraph/src/annotation/binary_matrix/column_sparse/column_major.cpp

```diff
@@ -76,6 +76,17 @@ ColumnMajor::slice_rows(const std::vector<Row> &row_ids) const {
     return slice;
 }
 
+void ColumnMajor::call_columns(const std::vector<Column> &columns,
+                               const std::function<void(Column, bitmap&&)> &callback,
+                               size_t num_threads) const {
+    #pragma omp parallel for num_threads(num_threads) schedule(dynamic)
+    for (size_t j = 0; j < columns.size(); ++j) {
+        const bitmap *c = (*columns_)[columns[j]].get();
+        callback(j, bitmap_generator([c](const auto &call_one) { c->call_ones(call_one); },
+                                     c->size(), c->num_set_bits()));
+    }
+}
```

Comment:
The method is const, but you call an rvalue, hence you must either copy or create a wrapper on every call.
Making it const reference `const bitmap &` will be better: you can still copy or wrap it in the caller, but in addition to this, you can call the column directly, as would be useful in this case:
```suggestion
void ColumnMajor::call_columns(const std::vector<Column> &columns,
                               const std::function<void(Column, const bitmap &)> &callback,
                               size_t num_threads) const {
    #pragma omp parallel for num_threads(num_threads) schedule(dynamic)
    for (size_t j = 0; j < columns.size(); ++j) {
        callback(j, *(*columns_)[columns[j]]);
    }
}
```